File: Practica8/Principal.py

class A:
    def __init__(self, x, z):
        self.x = x
        self.z = z

    def incrementaX(self):
        self.x += 1

    def incrementaZ(self):
        self.z += 1

class B:
    def __init__(self, y, z):
        self.y = y
        self.z = z

    def incrementaY(self):
        self.y += 1

    def incrementaZ(self):
        self.z += 1

class D(A, B):
    def __init__(self, x, y, z):
        A.__init__(self, x, z)
        B.__init__(self, y, z)

    def incrementaXYZ(self):
        self.x += 1
        self.y += 1
        self.z += 1

objeto_d = D(10, 20, 30)
print(f"Valores iniciales: x={objeto_d.x}, y={objeto_d.y}, z={objeto_d.z}")

objeto_d.incrementaX()
objeto_d.incrementaY()
objeto_d.incrementaZ()
print(f"Valores despues de incrementos individuales: x={objeto_d.x}, y={objeto_d.y}, z={objeto_d.z}")

objeto_d.incrementaXYZ()
print(f"Valores despues de incrementaXYZ: x={objeto_d.x}, y={objeto_d.y}, z={objeto_d.z}")

File: Practica8/Principal.java
package Practica8;

class A {
    int x;
    int z;

    public A(int x, int z) {
        this.x = x;
        this.z = z;
    }

    public void incrementaX() {
        this.x++;
    }

    public void incrementaZ() {
        this.z++;
    }
}

class B {
    int y;
    int z;

    public B(int y, int z) {
        this.y = y;
        this.z = z;
    }

    public void incrementaY() {
        this.y++;
    }

    public void incrementaZ() {
        this.z++;
    }
}

class D extends A {
    B b;

    public D(int x, int y, int z) {
        super(x, z);
        this.b = new B(y, z);
    }

    public void incrementaXYZ() {
        super.incrementaX();
        this.b.incrementaY();
        super.incrementaZ();
    }

    public int getY() {
        return this.b.y;
    }

    public int getZ() {
        return super.z;
    }
}

public class Principal {
    public static void main(String[] args) {
        D objetoD = new D(10, 20, 30);
        System.out.println("Valores iniciales: x=" + objetoD.x + ", y=" + objetoD.getY() + ", z=" + objetoD.getZ());

        objetoD.incrementaX();
        objetoD.b.incrementaY(); 
        objetoD.incrementaZ();   
        System.out.println("Valores despues de incrementos individuales: x=" + objetoD.x + ", y=" + objetoD.getY() + ", z=" + objetoD.getZ());

        objetoD.incrementaXYZ();
        System.out.println("Valores despues de incrementaXYZ: x=" + objetoD.x + ", y=" + objetoD.getY() + ", z=" + objetoD.getZ());
    }
}
